packages/typespec-ts/src/index.ts

```diff
@@ -403,8 +403,7 @@ export async function $onEmit(context: EmitContext) {
     );
     const hasPackageFile = await existsSync(existingPackageFilePath);
     const shouldGenerateMetadata =
-      option.generateMetadata === true ||
-      (option.generateMetadata === undefined && !hasPackageFile);
+      option.generateMetadata === true || !hasPackageFile;
```

Comment:
I think this broke the scenario where a user explicitly passes `false` to generate-metadata which should be honored. Keyvault is a prime example of this https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/keyvault/keyvault-admin/package.json#L45